src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
-package core.basesyntax;
-
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
-
-    }
-
-    @Override
-    public void add(T value, int index) {
-
-    }
-
-    @Override
-    public void addAll(List<T> list) {
-
-    }
-
-    @Override
-    public T get(int index) {
-        return null;
-    }
-
-    @Override
-    public void set(T value, int index) {
-
-    }
-
-    @Override
-    public T remove(int index) {
-        return null;
-    }
-
-    @Override
-    public T remove(T element) {
-        return null;
-    }
-
-    @Override
-    public int size() {
-        return 0;
-    }
-
-    @Override
-    public boolean isEmpty() {
-        return false;
-    }
-}
+package core.basesyntax;
+
+import java.util.NoSuchElementException;
+
+public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private int cursor;

```

Comment:
cursor it`s something like size or length?

Reply:
Yes. Its purpose is as length.